python3Packages.reinkpy: init at 2.240918.1-unstable-2025-04-28

Luflosi opened the pull request:
I successfully used this package to reset the waste ink counter of my EPSON ET-2720 printer.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Luflosi committed:
python3Packages.reinkpy: init at 2.240918.1-unstable-2025-04-28

Sigmanificient commented:
getting the following error:
```py
WARNING reinkpy.usb      No module named 'usb'
WARNING reinkpy.zeroconf No module named 'zeroconf'
Traceback (most recent call last):
  File "<python-input-4>", line 2, in <module>
    for p in reinkpy.Device.find(): print(p)
             ~~~~~~~~~~~~~~~~~~~^^
  File "/home/sigmanificient/.cache/nixpkgs-review/pr-405652-1/results/python313Packages.reinkpy-x86_64-linux/lib/python3.13/site-packages/reinkpy/__init__.py", line 81, in find
    res = [d for c in cls.__subclasses__() for d in c.ifind(timeout=timeout)]
                                                    ~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/sigmanificient/.cache/nixpkgs-review/pr-405652-1/results/python313Packages.reinkpy-x86_64-linux/lib/python3.13/site-packages/reinkpy/__init__.py", line 135, in ifind
    from .zeroconf import find
  File "/home/sigmanificient/.cache/nixpkgs-review/pr-405652-1/results/python313Packages.reinkpy-x86_64-linux/lib/python3.13/site-packages/reinkpy/zeroconf.py", line 21, in <module>
    class Browser:
    ...<24 lines>...
                    d.pop(a, None)
  File "/home/sigmanificient/.cache/nixpkgs-review/pr-405652-1/results/python313Packages.reinkpy-x86_64-linux/lib/python3.13/site-packages/reinkpy/zeroconf.py", line 35, in Browser
    def on_change(self, zeroconf: Zeroconf, service_type: str, name: str,
                                  ^^^^^^^^
NameError: name 'Zeroconf' is not defined
```

Luflosi commented:
Thank you very much for noticing!

I only tested this package with `pysnmp` and `zeroconf` added to the `dependencies` list but since I couldn't get that to work and the packages are marked as optional, I commented them out. I guess they are not so optional after all.

Now I added them back and added a `longDescription` so it is more likely that users are aware of the issue.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5507

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2436
